test/stackTrace.js

```diff
@@ -98,6 +104,56 @@ tap.test('parses function name from original stack for anonymous function', func
     });
 });
 
+tap.test('parses function info from original stack for Promise scenario', function (tt) {
+    tt.plan(4);
+
+    var test = tape.createHarness();
+    test.createStream();
+
+    test._results._watch = function (t) {
+        t.on('result', function (res) {
+            tt.equal('onfulfilled', res.functionName);
+            tt.match(res.file, /stackTrace.js/i);
+            tt.ok(Number(res.line) > 0);
+            tt.ok(Number(res.column) > 0);
+        });
+    };
+
+    test('t.equal stack trace', function testFunctionNameParsing(t) {
+        new Promise(function (resolve) {
+            resolve();
+        }).then(function onfulfilled() {
+            t.equal(true, false, 'true should be false');
+            t.end();
+        });
+    });
+});
+
+tap.test('parses function info from original stack for Promise scenario with anonymous function', function (tt) {
```

Comment:
this test needs to be skipped when `typeof Promise !== 'function'`

Reply:
Done